### PR TITLE
Integrating depth anything 3

### PR DESCRIPTION
**Pull request by manas1245agrawal**
## 📝 Description

**⚠️ Issue Link Required**: This PR must be linked to an approved and assigned issue. See [Contributing Guide](https://github.com/kornia/kornia/blob/main/CONTRIBUTING.md#pull-request) for details.

Issue link - https://github.com/kornia/kornia/issues/3488

**Fixes/Relates to:** #  3488

**Important**:
- Ensure you are assigned to the linked issue before submitting this PR
- This PR should strictly implement what the linked issue describes
- Do not include changes beyond the scope of the linked issue

---

## 🛠️ Changes Made
- [x] Integrated Depth anything 3 model
- [x] Added Tests for the same
- [x] Added pose estimation metric for the same

---

## 🧪 How Was This Tested?
- [x] **Unit Tests:** (List new/updated tests)
- [x] **Manual Verification:** (Describe the steps you took)
- [ ] **Performance/Edge Cases:** (How does this handle nulls, large data, etc.?)

---

## 🕵️ AI Usage Disclosure
*Check one of the following:*
- [ ] 🟢 **No AI used.**
- [x] 🟡 **AI-assisted:** I used AI for boilerplate/refactoring but have manually reviewed and tested every line.
- [ ] 🔴 **AI-generated:** (Note: These PRs may be subject to stricter scrutiny or immediate closure if the logic is not explained).

---

## 🚦 Checklist
- [x] I am assigned to the linked issue (required before PR submission)
- [x] The linked issue has been approved by a maintainer
- [x] I have performed a **self-review** of my code (no "ghost" variables or hallucinations).
- [x] My code follows the existing style guidelines of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] (Optional) I have attached screenshots/recordings for UI changes.

---

### CODE REVIEW

**Comment by Borda** [comment]
Is it copy-paste code? Why not just refer to the package or git sub-module?

**Comment by manas1245agrawal** [comment]
> Is it copy-paste code? Why not just refer to the package or git sub-module?

Sir, I have inherited from the main repo indeed. The reason for not referring is because maintainer preferred to have something that future researchers can work and fine-tune on. I believe for working and making a good library, it is better to have the model directly integrated instead of calling for an API.

**Comment by Borda** [comment]
anyaway seems you started working on an issue you created yourself without any warning from mainners or any other approval? 🤔

**Comment by manas1245agrawal** [comment]
> anyaway seems you started working on an issue you created yourself without any warning from mainners or any other approval? 🤔

No sir, I informed the maintainer beforehand about around 15 days ago, its just I created the issue yesterday

**Comment by edgarriba** [comment]
this is massive PR, we cannot accept in this shape. See how other models are imported step by step  without utilities and clear model definition, as we intend to have clean and not convoluted implementations. Only pytorch is allowed.

**Comment by manas1245agrawal** [comment]
> this is massive PR, we cannot accept in this shape. See how other models are imported step by step without utilities and clear model definition, as we intend to have clean and not convoluted implementations. Only pytorch is allowed.

Okay I will break into multiple PRs. Should I first write a proper proposal in issue

**Comment by edgarriba** [comment]
> > this is massive PR, we cannot accept in this shape. See how other models are imported step by step without utilities and clear model definition, as we intend to have clean and not convoluted implementations. Only pytorch is allowed.
> 
> Okay I will break into multiple PRs. Should I first write a proper proposal in issue

yes

**Comment by manas1245agrawal** [comment]
> > > this is massive PR, we cannot accept in this shape. See how other models are imported step by step without utilities and clear model definition, as we intend to have clean and not convoluted implementations. Only pytorch is allowed.
> > 
> > 
> > Okay I will break into multiple PRs. Should I first write a proper proposal in issue
> 
> yes

Good evening, I have given a detailed proposal on how I will be integrating the model 

https://github.com/kornia/kornia/issues/3488

Please give the feedback, if you agree with my initial plan, I can start right away
